### DISCUSSION TITLE
More Classifiers about Sklearn? SVM?

### DISCUSSION THREAD

**Discussion by AiceQueen** [autogluon/autogluon#1728] [Q&A] [answered]
Hello, I am so glad to talk with you. I'm a student and a primary learner about ML. So, I want to know how to use other classifiers in my Tabular Prediction Project. According to your AutoGluon Tasks, I can successfully use your classifiers you have mentioned such as XGBoost, LGBM and RF. Some of these are based on sklearn, then I also want to use SVC from sklearn. Can that be done? And how? SVM doesn't be mentioned in your parameters probably, I think. 
Thanks, have a good time.

**Comment by Innixma** [comment, marked as answer]
If you'd like to use a model that isn't already provided in AutoGluon, you can refer to the custom model tutorial: https://auto.gluon.ai/stable/tutorials/tabular_prediction/tabular-custom-model.html

Generally speaking, adding sklearn models via the custom model tutorial should be very easy, you can mostly re-use the CustomRandomForestModel code, but instead use SVM in this section:

```
        if self.problem_type in ['regression', 'softclass']:
            model_cls = RandomForestRegressor
        else:
            model_cls = RandomForestClassifier
```

Note that usually if a model is not part of AutoGluon's default, it is because it is not performant compared to the existing models, and so it will be unlikely to improve AutoGluon's ensembled result if added.
